thirdeye/thirdeye-frontend/app/pods/manage/alert/explore/route.js

```diff
@@ -423,6 +366,77 @@ export default Route.extend({
     }
   },
 
+  /**
+   * Fetches all anomaly data for found anomalies - downloads all 'pages' of data from server
+   * in order to handle sorting/filtering on the entire set locally. Start/end date are not used here.
+   * @param {Array} anomalyIds - list of all found anomaly ids
+   * @returns {RSVP promise}
+   */
+  fetchCombinedAnomalies: task(function * (anomalyIds) {
+    yield timeout(300);
+    if (anomalyIds.length) {
+      const idGroups = toIdGroups(anomalyIds);
+      const anomalyPromiseHash = idGroups.map((group, index) => {
+        let idStringParams = `anomalyIds=${encodeURIComponent(idGroups[index].toString())}`;
+        let url = `/anomalies/search/anomalyIds/0/0/${index + 1}?${idStringParams}`;
+        let getAnomalies = this.get('fetchAnomalyEntity').perform(url);
+        return RSVP.resolve(getAnomalies);
+      });
+      return RSVP.all(anomalyPromiseHash);
+    } else {
+      return RSVP.resolve([]);
+    }
+  }),
+
+  /**
+   * Fetches change rate data for each available anomaly id
+   * @method fetchCombinedAnomalyChangeData
+   * @param {Array} anomalyData - array of processed anomalies
+   * @returns {RSVP promise}
+   */
+  fetchCombinedAnomalyChangeData: task(function * (anomalyData) {
+    yield timeout(300);
+    let promises = [];
+
+    anomalyData.forEach((anomaly) => {
+      let id = anomaly.anomalyId;
+      promises[id] = this.get('fetchAnomalyEntity').perform(`/anomalies/${id}`);
```

Comment:
are we assigning at index id here? Seems like an object would be more correct. Any thought?

Reply:
This promise resolves with the anomaly Id as associative array so that I can filter it if needed later.